lib/modem_info/modem_info.c

```diff
@@ -71,6 +78,9 @@ LOG_MODULE_REGISTER(modem_info);
 #define ICCID_PARAM_INDEX 2
 #define ICCID_PARAM_COUNT 3
 
+#define SYSTEMMODE_PARAM_INDEX 0
+#define SYSTEMMODE_PARAM_COUNT 4
+
 #define CMD_SIZE(x) (strlen(x) - 1)
```

Comment:
CMD_SIZE as implemented here give false expectations of what the macro actually do. It doesn't give the CMD_SIZE, but the CMD_SIZE - 1. Maybe SKIP_CMD/GET_PAYLOAD would be a better name?

I also want to discuss a bit how it's implemented:

Example:
`#define AT_CMD_FOOBAR "AT+FOOBAR"`

`strlen(AT_CMD_FOOBAR) - 1`

 will return 8, so the number will be what you want, but it is based on false premises. Can we trust that it always will be like this? Or would be better to change the approach a bit. 

Maybe change skip AT%/AT+ in the defines like this:

`#define AT_CMD_FOOBAR  "FOOBAR"`

Then use a some macros to do what you want

#define SET_AT_CMD(cmd, param) "AT+"cmd"="param you will need another one for proprietary commands or have a third parameter to insert either "%" or "+". 
#define READ_AT_CMD(cmd) "AT+"cmd"?"
#define TEST_AT_CMD(cmd) "AT+"cmd"=?" if you need that

Then you could also do find a
#define GET_PAYLOAD(cmd) strlen(cmd) + 2, the value 2 should also probably be a define with a readable name.

Just a thought, it think it will make the code more readable/maintainable

Reply:
Oh yes, I agree. Thank you for your input, it is very educational for me.

The CMD_SIZE() macro is actually not used anymore and should be removed, I simply forgot. I've decided to create the function modem_info_remove_cmd(), I believe it is better suited for my purpose. You're welcome to add any input to this usage if you'd like.